eth/api.go

```diff
@@ -636,3 +636,33 @@ func storageRangeAt(st state.Trie, start []byte, maxResult int) StorageRangeResu
 	}
 	return result
 }
+
+func (api *PrivateDebugAPI) GetDirtyAccountsByNumber(startNum, endNum uint64) ([]common.Address, error) {
+	return api.getDirtyAccounts(api.eth.blockchain.GetBlockByNumber(startNum), api.eth.blockchain.GetBlockByNumber(endNum))
+}
+
+func (api *PrivateDebugAPI) GetDirtyAccountsByHash(startHash, endHash common.Hash) ([]common.Address, error) {
+	return api.getDirtyAccounts(
+		api.eth.blockchain.GetBlockByHash(startHash),
+		api.eth.blockchain.GetBlockByHash(endHash))
+}
+
+func (api *PrivateDebugAPI) getDirtyAccounts(startBlock, endBlock *types.Block) ([]common.Address, error) {
```

Comment:
I'd suggest adding a check to ensure startBlock is before endBlock to not be surprising that the inverted parametrization also works.

Also perhaps it would be nice to check that the blocks are on the same chain, though that may be expensive for large chains, so I'm uncertain about that.

Reply:
> Also perhaps it would be nice to check that the blocks are on the same chain, though that may be expensive for large chains, so I'm uncertain about that.

Actually, now that you mention it, it can be another interesting usecase for this; investigating if there double-spends or generally diffs between non-canon blocks and canon-blocks. If the functionality is kept as is.